File: CompilationTest/ExceptionTest/results/ExceptionTest.cs
using System;

public class TestClass
{
    public int NotThrows()
    {
        return 5;
    }
    
    public void FThrows()
    {
        throw new Exception("exception message");
    }
    
    public void TestMethod()
    {
        Console.WriteLine(this.NotThrows());
        this.FThrows();
    }
}

public class Program
{
    static public void Main(string[] args)
    {
        try 
        {
            new TestClass().TestMethod();
        }
        catch (System.Exception e)
        {
            System.Console.WriteLine($"Exception: {e.Message}");
        }
    }
}

File: CompilationTest/ExceptionTest/results/ExceptionTest.cpp
#include <iostream>
#include <stdexcept>

class TestClass {
  public:
    int notThrows() {
        return 5;
    }
    
    void fThrows() {
        throw std::runtime_error(std::string("exception message"));
    }
    
    void testMethod() {
        std::cout << this->notThrows() << std::endl;
        this->fThrows();
    }

  private:
};

int main()
{
    try {
        TestClass c;
        c.testMethod();
    } catch(std::exception& err) {
        std::cout << "Exception: " << err.what() << '\n';
    }
    return 0;
}